### PR TITLE
feat(consistent-data-testid): add support for custom error message

### LINKED ISSUE #718: consistent-data-testid: add support for custom error message
### What rule do you want to change?

consistent-data-testid

### Does this change cause the rule to produce more or fewer warnings?

More warnings

### How will the change be implemented?

The changes will be implemented by extending the possible rule configuration values by a new field (e.g. `customMessage`), and consuming it, if present.

### Example code

Example ESLint config:

```js
'testing-library/consistent-data-testid': [
    'error',
    {
        'testIdPattern': '^([a-z][a-z0-9]*)(-[a-z0-9]+)*$', // kebab-case
        'testIdAttribute': ['data-testid'],
    }
]
```

```html
<div data-testid="some_snake_case_value" />
```

### How does the current rule affect the code?

Example ESLint config:

```js
'testing-library/consistent-data-testid': [
    'error',
    {
        'testIdPattern': '^([a-z][a-z0-9]*)(-[a-z0-9]+)*$', // kebab-case
        'testIdAttribute': ['data-testid'],
    }
]
```

Example code:
```html
<div data-testid="some_snake_case_value" />
```

Example error: 

```
ESLint: `data-testid` "some_snake_case_value" should match '^([a-z][a-z0-9]*)(-[a-z0-9]+)*$'' (testing-library/consistent-data-testid)
```

### How will the new rule affect the code?

Example ESLint config:

```js
'testing-library/consistent-data-testid': [
    'error',
    {
        'testIdPattern': '^([a-z][a-z0-9]*)(-[a-z0-9]+)*$', // kebab-case
        'testIdAttribute': ['data-testid'],
        'customMessage': 'Please use kebab-case values for data-testids.'
    }
]
```

Example code:
```html
<div data-testid="some_snake_case_value" />
```

Example error: 

```
ESLint: `data-testid` Please use kebab-case values for data-testids. (testing-library/consistent-data-testid)
```

### Anything else?

First things first, I hope I've used the right issue template. Also, I am more than happy to have a go at this issue and open a PR, in case you all agree to proceed with this change.

In my opinion adding this change would support devs in providing more meaningful (error) messages. In my example, the developer does not need to think about what the regex means, but is just shown a straight-forward message.

Any feedback is welcome :) 

### Do you want to submit a pull request to change the rule?

Yes

### PR DESCRIPTION
**Pull request by patriscus**
## Checks

- [x] I have read the [contributing guidelines](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/CONTRIBUTING.md).

## Changes

- Add `customMessage` as an optional field for the `Options`
- Use `''` as the default for `customMessage`
- Add a new message id, used to display the custom message
- Add a function to determine which message id to use, based on `customMessage`'s value
- Pass `customMessage` along in the `data` object when calling `context.report()`
- Add test case
- Update docs

## Context

Fixes #718

### CODE REVIEW

**Review by Belco90 — COMMENTED** [review]
Hi @patriscus! Thanks for your PR, it looks great in general. Just a couple of minor things and should be ready to go.

File: lib/rules/consistent-data-testid.ts [hunk 1]
```diff
@@ -100,6 +109,18 @@ export default createTestingLibraryRule<Options, MessageIds>({
...
 		}
 
+		function getErrorMessageId(): MessageIds {
+			if (customMessage === undefined) {
+				return 'consistentDataTestId';
+			}
+
+			if (customMessage === '') {
+				return 'consistentDataTestId';
+			}
```
[the remark this answers is not in the document]
Comment (patriscus) [reply lib/rules/consistent-data-testid.ts]: I agree with you - changed the `default` to `undefined` and removed the `customMessage === ''` check now 👍

File: tests/lib/rules/consistent-data-testid.test.ts [hunk 2]
```diff
@@ -409,6 +413,38 @@ const invalidTestCases: InvalidTestCase[] = [
 					attr: 'data-testid',
 					value: 'WrongComponent__cool',
 					regex: '/^Parent(__([A-Z]+[a-z]*?)+)*$/',
+					message: '',
+				},
+			},
+		],
+	},
+	{
+		code: `
```
[the remark this answers is not in the document]
Comment (patriscus) [reply tests/lib/rules/consistent-data-testid.test.ts]: Done :)

**Review by Belco90 — APPROVED** [review]

**Comment by Belco90** [comment]
@all-contributors please add @patriscus for ideas, code, test

**Pull request merged**